0000-report-inboxes.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: report-inboxes
+- Start Date: 2024-02-20
+- RFC PR: [LemmyNet/rfcs#0000](https://github.com/LemmyNet/rfcs/pull/0000)
+- Lemmy Issue: [LemmyNet/lemmy#0000](https://github.com/LemmyNet/lemmy/issues/0000)
+
+# Summary
+
+Rather than combining all reports into a single report inbox, we should allow users to select whether they are reporting to mods or admins, and we should split reports into different inboxes based on that selection.
+
+# Motivation
+
+The current approach has some shortcomings:
+
+* Users are not currently able to bypass mods and report directly to admins - this may allow mods to conceal instance rule breaking in specific communities
+* Admins are not aware of community rules, so they may wish to take no action for most community rule breaking reports. However, if an admin resolves such a report, the relevant community mods most likely never see it.
+* Different instances may have different rules, but somebody resolving a report on one instance will resolve it for other instances as well, thus potentially resulting in missed reports.
+* Mods might take local action on a report and mark it as resolved even in cases where a user should be banned from the entire instance. In this case, admins are very unlikely to see the report.
+
+# Guide-level explanation
+
+### When creating reports, users will be able to select if it's a mod report, or an admin report (or both)
+
+![image](https://github.com/sunaurus/lemmy-rfcs/assets/5356547/9a21b527-6c88-4024-b287-3371d77688f4)
+
+### Instead of the current single report inbox, there will be three different kinds of inboxes
+
+* Admin reports - show all reports sent to admins (only visible to admins)
+* Mod reports - show all reports sent to mods for any communities the user moderates
+* All reports - Shows a read-only view of all (admin and mod) reports, only visible to admins
+   * This is a read-only version of the current 0.19.3 admin report view, and would allow admins to still keep an eye on mod actions on their instance if they wish
+    
+The UI wouldn't need to change for mods, but for admins, there would be a new selection at the top of the reports page (the "mod reports" tab would only be visible if the admin is also a mod in any community):
+![image](https://github.com/sunaurus/lemmy-rfcs/assets/5356547/cc4ad68c-6e85-4cd9-b324-131c06951cb3)
+
+### Resolving reports should be more granular
+
+* Reports in the "admin reports" tab can only be resolved for admins of the local instance
+   * To reduce overhead, **banning the reported user on the user's home instance + removing reported content should automatically resolve reports for remote admins as well**
```

Comment:
I don't understand this - what do you mean by remove admins? I was under the impression that admin reports only go to the local admin and doesn't federate. Why would you want to report something to an admin that is not of your own instance?

Do we need 3 options? I.e.:

1. Report to community mods (regardless of whether they are local or remote users).
2. Report to local admins (*only* my own instance's admins)
3. Report to _all_ admins (federate report to all admins the instance knows of)

That last options seems... excessive? I suspect I could be flooded with reports even if I only had a small instance. Is the "report to admins" you're talking about in this RFC number 2 or number 3?

Reply:
I think users should not have the ability to generate reports to ALL federated admins, however, I think when submitting a report to admins, it should always go to all relevant admins.

The set of relevant admins depends on context, there are three options:

1. feddit.dk user reports a post by another feddit.dk user, which was created in a feddit.dk community 
    * Only feddit.dk admins should get the report
2. feddit.dk user reports a post by a lemm.ee user in either a lemm.ee or a feddit.dk community
    * Both lemm.ee and feddit.dk admins should receive the report
3. feddit.dk user reports a post by a lemm.ee user in a lemmy.ml community
    * All of lemm.ee, feddit.dk and lemmy.ml admins should get the report

In the second and third scenario, it makes sense to automatically resolve reports for remote admins as well, when the user is banned on their home community & the reported content has been removed (because at that point there is no further actions any admin on any instance can take for that report).

However, if a feddit.dk admin bans the lemm.ee user first, the report should not be automatically resolved for lemm.ee admins, because they might still want to ban the user locally as well.

Reply:
That makes a lot of sense - I don't think the way the RFC is written right now makes it clear that that is what you mean, perhaps explicitly talk about those 3 scenarios as it made it clear to me :)